layered_diffusion.py

```diff
@@ -68,14 +74,17 @@ def decode(self, samples, images):
                     else torch.float32
                 ),
             )
-        latent = samples["samples"]
         pixel = images.movedim(-1, 1)  # [B, H, W, C] => [B, C, H, W]
-        pixel_with_alpha = self.vae_transparent_decoder.decode_pixel(pixel, latent)
+        decoded = []
+        for start_idx in range(0, samples["samples"].shape[0], per_batch):
+            decoded.append(self.vae_transparent_decoder.decode_pixel(pixel[start_idx:start_idx+per_batch], samples["samples"][start_idx:start_idx+per_batch]))
+        pixel_with_alpha = torch.cat(decoded, dim=0)
+        
         # [B, C, H, W] => [B, H, W, C]
         pixel_with_alpha = pixel_with_alpha.movedim(1, -1)
         image = pixel_with_alpha[..., 1:]
         alpha = pixel_with_alpha[..., 0]
-        return (image, alpha)
+        return (image.cpu(), alpha.cpu(),)
```

Comment:
Why do we need to move tensors to cpu here?

Reply:
That is not necessary indeed, there was some node that expected it on cpu when I was testing, removed it now.

Reply:
They do seem necessary though. See https://github.com/huchenlei/ComfyUI-layerdiffuse/issues/28.

Reply:
Honestly there doesn't seem to be a consensus which way it's supposed to be in Comfy, I do think .cpu() can be safer as most stuff that need it in gpu convert it first anyway.